### PR TITLE
Add MetricEstimatesNext support

### PR DESCRIPTION
**Pull request by jessie-ross**
**This PR adds MetricEstimatesNext support and transforms it to MetricEstimatesPrevious:**
- Also renames MetricEstimates and Analysis to MetricEstimatesPrevious and AnalysisPrevious.

The counterpart of 139-gh-Automattic/a8cds, it should be merged first.

Part of A/B/n and multiple credible intervals: 
- 661-gh-Automattic/experimentation-platform
- 443-gh-Automattic/experimentation-platform

This is a transformational refactor where we have a boundary - the point of transformation of AnalysisPrevious and AnalysisNext - and slowly move that boundary down. This allows the changes to be easily decoupled and can move at the pace we need. It also allows us to do so in smaller easily split PRs. The boundary is currently at the very start in ETL and we have a PR 139-gh-Automattic/a8cds that will move the boundary from within the ETL to the API output level.

## How has this been tested?

- Automated tests that cover added/changed functionality

### CODE REVIEW

File: src/__tests__/api/utils.test.ts [hunk 1]
```diff
@@ -6,6 +6,12 @@ import { fetchApi } from 'src/api/utils'
 
 fetchMock.config.overwriteRoutes = true
 
+// In order to not go over API limits on swagger we wait in-between tests:
+const apiLimitWait = 500
+beforeAll(async () => {
+  return new Promise((resolve) => setTimeout(resolve, apiLimitWait))
+})
+
```
[the remark this answers is not in the document]
Comment (razvanpapadopol) [reply src/__tests__/api/utils.test.ts:14]: It is the right place if you want to add a delay before running (all) the tests in this file.
> Runs a function before any of the tests in this file run. If the function returns a promise or is a generator, Jest waits for that promise to resolve before running tests.

We just need to add it also to other files testing the api.

**Comment by jessie-ross** [comment]
Ah, I haven't actually done this part: 
> Also renames MetricEstimates and Analysis to MetricEstimatesPrevious and AnalysisPrevious.

File: src/lib/schemas.ts [hunk 2]
```diff
@@ -541,7 +541,34 @@ export const metricEstimateSchema = yup
...
   .camelCase()
-export interface MetricEstimate extends yup.InferType<typeof metricEstimateSchema> {}
+export interface MetricEstimatePrevious extends yup.InferType<typeof metricEstimatePreviousSchema> {}
+
+// It is a bit complicated to type this properly and we won't need it for long anyway so we take this shortcut:
+export type MetricEstimatesPrevious = Record<string | 'diff' | 'ratio', MetricEstimatePrevious>
+
+export const distributionStatsSchema = yup
+  .object({
+    mean: extendedNumberSchema.defined(),
```
Comment (razvanpapadopol) [inline src/lib/schemas.ts:551]: Just to confirm, this property is newly introduced and different than `estimate` which is [marked as "deprecated"](https://github.com/Automattic/abacus/blob/master/src/lib/schemas.ts#L527-L530), right?

File: src/lib/schemas.ts [hunk 3]
```diff
@@ -541,7 +541,34 @@ export const metricEstimateSchema = yup
   })
   .defined()
   .camelCase()
-export interface MetricEstimate extends yup.InferType<typeof metricEstimateSchema> {}
+export interface MetricEstimatePrevious extends yup.InferType<typeof metricEstimatePreviousSchema> {}
+
+// It is a bit complicated to type this properly and we won't need it for long anyway so we take this shortcut:
```
Comment (razvanpapadopol) [inline src/lib/schemas.ts:546]: 👍🏼

**Review by razvanpapadopol — APPROVED** [review]
While I haven't reviewed thoroughly the counterpart PR 139-gh-Automattic/a8cds, I find the schema changes solid and the new `metric_estimates` structure awesome! Looking forward to the next steps 🕺🏼

File: src/lib/schemas.ts (hunk 2 above)
Comment (jessie-ross) [reply src/lib/schemas.ts:551]: Yep!

File: src/__tests__/api/utils.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jessie-ross) [reply src/__tests__/api/utils.test.ts:14]: I've put it in each file and it seems to be working :)

**Pull request merged**